apis/python/src/tiledbsoma/indexed_dataframe.py

```diff
@@ -418,10 +415,10 @@ def _validate_schema(schema: pa.Schema, index_column_names: Sequence[str]) -> pa
     # verify that all index_column_names are present in the schema
     schema_names_set = set(schema.names)
     for index_column_name in index_column_names:
-        if index_column_name.startswith("soma_") and index_column_name != SOMA_JOINID:
```

Comment:
this check appeared to be redundant given that we're checking that the `index_column_names` are a subset of the `schema.names` where we made a similar check